systems/WikibaseRepo/05-Building_Block_View.md

```diff
@@ -1,3 +1,53 @@
 # Building Block View
 
-## Whitebox Overall System
\ No newline at end of file
+## Whitebox Overall System
+
+TBA
+
+### External Interfaces
+
+#### Action API Edit Entity
+
+![Alt Text](./diagrams/05-api-wbeditentity.drawio.svg)
+
+|  Building Block | Type/Context  | Responsibility   |
+| ------------ | ------------ | ------------ |
+| ModifyEntity   |  base class | Base class for API modules modifying a single entity  |
+| EntityIdParser | Handling user input | Interface for objects that can parse strings into EntityIds  |
+| StringNormalizer  | Handling user input   | Normalizes strings  |
+| SiteLinkTargetProvider  | Handling user input  |  Returns the list of sites that is suitable as a sitelink target |
+| enabledEntityTypes  | configuration | List of entity type identifiers (typically "item" and "property") that are configured in WikibaseRepo.entitytypes.php  |
+| badgeItems  | configuration | Items allowed to be used as badges  |
+| sitelinkGroups  | configuration | The site groups to use in sitelinks  |
+| ContentLanguages  | configuration | A list of languages supported as content language  |
+| propertyDataTypes  | configuration |  A list of property data types |
+| $federatedPropertiesEnabled  | configuration | Is federated properties enabled on the wikibase  |
+| EntityChangeOpProvider  | Changes applied to the entity | Turns entity change request into ChangeOp objects based on change request deserialization configured for the particular entity type  |
+|  ChangeOpDeserializers | Changes applied to the entity | ChangeOpDeserializers for fields of items and properties, such as label, description, alias, claim and sitelink  |
+|  ChangeOps | Changes applied to the entity | Class for holding a batch of change operations  |
+| TitleLookup  | lookups |   Represents an arbitrary mapping from entity IDs to wiki page titles, with no further guarantees given. The resulting title does not necessarily represent the page that actually stores the entity contents. |
```

Comment:
The interface is `EntityTitleLookup` so perhaps we should say that here.
I noticed this as while going through https://github.com/wmde/wikidata-wikibase-architecture/pull/18 I added `EntityTitleLookup` to a table with the same description as the `TitleLookup` here